docs/proposals/ingress.md

```diff
@@ -0,0 +1,91 @@
+# Ingress Support at Edge
+
+
+## Abstract:
+
+To provide [ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) support at the edge. Ingress exposes HTTP routes from outside the cluster to services within the cluster. 
+Using ingress the user can bring all their services under a single umbrella and manage them easily. The ingress support should continue to work even in offline scenarios i.e. even in the cases where 
+connectivity between edge and cloud is lost/unreliable. 
+
+
+## Motivation:
+
+The cloud native and microservice architecture is becoming more and more popular and the edge node is becoming more and more powerful. In these scenarios, users might want to deploy their applications at the edge, 
+which could need to be accessed by external users, through services. Using ingress we would be able to provide name based virtual hosting and load balance traffic.
+
+This design document deals with enabling users to leverage powerful machines at the edge to provide external users 
+access to the services running within the different edge nodes.  
+
+
+## Goals:
+
+- Provide external users access to the multiple services running within the edge nodes in the same cluster.
+- Provide name based virtual hosting and load balance traffic.
+- Provide ingress support even in offline scenarios
+
+
+## Non-Goals:
+
+- To provide access to services running in different clusters
+
+
+## Proposal:
+We propose to use [kubernetes ingress resources](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource) to connect to various services which in turn connect to their application.
+We are proposing to use an [Nginx ingress controller](https://www.nginx.com/products/nginx/kubernetes-ingress-controller) (Other [ingress controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/#additional-controllers) 
+could also be utilized for this purpose, but for the current scope we are only considering a Nginx ingress controller) which would route the user requests to the respective service. 
+Traffic routing is controlled by rules defined on the Ingress resource. This proposal uses Node-port service to redirect external users to the ingress service, instead we can also make use of an external load-balancer for this purpose. 
+
+
+## Design Details:
+
+In the current design the communication between external users and the services are handled by the nginx ingress controller. The communication between the services and their respective application pods will be handled by the edge mesh feature of KubeEdge. 
+
+<img src="../images/ingress/ingress.png">
+
+We are making use of nginx controller as our ingress controller to handle ingress resources. The nginx ingress controller runs as a pod (which would be deployed by edge core) in its own namespace (ingress-nginx). 
```

Comment:
currently edge_core only support pods from 1 namespace.... if the namespace is ingress-nginx, we may not be able to support pods from multiple namespaces ?

Reply:
I think we already have an issue to support multiple pod namespace https://github.com/kubeedge/kubeedge/issues/943. Once that feature is ready i don't think it would be a problem if all resources used by ingress controller resides in a separate namespace. Thoughts ?